lib/stitches/linear_gradient_fill.py

```diff
@@ -0,0 +1,241 @@
+# Authors: see git history
+#
+# Copyright (c) 2010 Authors
+# Licensed under the GNU GPL version 3.0 or later.  See the file LICENSE for details.
+
+from math import ceil, floor, sqrt
+
+import numpy as np
+from inkex import DirectedLineSegment, Transform
+from shapely import segmentize
+from shapely.affinity import rotate
+from shapely.geometry import LineString, MultiLineString, Point, Polygon
+
+from ..svg import get_correction_transform
+from ..utils.threading import check_stop_flag
+from .auto_fill import (build_fill_stitch_graph, build_travel_graph, fallback,
+                        find_stitch_path, graph_is_valid)
+from .circular_fill import path_to_stitches
+from .guided_fill import apply_stitches
+
+
+def linear_gradient_fill(fill, shape, starting_point, ending_point):
+    has_gradient, colors, lines, line_nums = _get_lines_nums_and_colors(shape, fill)
+    color_lines, colors = _get_color_lines(colors, line_nums, lines)
+    if not has_gradient:
+        colors.pop()
+    stitch_groups = _get_stitch_groups(fill, shape, colors, color_lines, starting_point, ending_point)
+    return stitch_groups
+
+
+def _get_lines_nums_and_colors(shape, fill):
+    # min x, min y, max x, max y
+    orig_bbox = shape.bounds
+
+    has_gradient = True
+    if "linearGradient" not in fill.color:
+        # they have no linear gradient, let's simply space out the rows
+        has_gradient = False
+        angle = 0
+        offsets = [0, 1]
+        colors = [fill.color, 'none']
+        point1 = (orig_bbox[0], orig_bbox[1])
+        point2 = (orig_bbox[2], orig_bbox[3])
+    else:
+        node = fill.node
+        color = fill.color[5:-1]
+        xpath = f'.//svg:defs/svg:linearGradient[@id="{color}"]'
+        gradient = node.getroottree().getroot().findone(xpath)
+        gradient.apply_transform()
+
+        offsets = gradient.stop_offsets
+        colors = [style['stop-color'] for style in gradient.stop_styles]
+        transform = -Transform(get_correction_transform(node, child=True))
```

Comment:
I think `get_node_transform(node)` will work here.  `get_correction_transform(node, child=True)` is just `-get_node_transform(node)`.

Reply:
Oh, ok. I'll try that out then.